### PR TITLE
KUTTL Integration tests

### PR DESCRIPTION
**Pull request by yaksharma**
Runs integration test using KUTTL

## Description

Runs KUTTL TestSuite on a local Kind cluster using mock prometheus metrics provided by victoria metrics, for testing scale up and scale down functionality using KratosSpec in a configmap. 

## Motivation and Context

Adds Integration tests.

## How Has This Been Tested?

Tested using a local kind cluster.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: scale/scale_facade.go [hunk 1]
```diff
@@ -212,6 +212,8 @@ func (f *ScaleFacade) calculateMaxScaleReplicas(item *corev1.ConfigMap, currentR
 
 		log.V(1).Info("metric values and replica proposal", "replicas", replicaProposal, "metrics", metricValues)
 
+		log.V(1).Info("metric values and replica proposal", "replicas", replicaProposal, "metrics", metricValues)
```
Comment (amuraru) [inline scale/scale_facade.go:215]: This seems duplicate?

File: magefile.go [hunk 2]
```diff
@@ -22,18 +22,26 @@ import (
...
 
-// Cleans project
+// Clean Cleans project
 func Clean() {
 	//mg.Deps(operator.Clean)
 	mg.Deps(operator.Clean, helm.Clean)
 }
 
-// Build code, docker image and helm chart
+// BUild Builds code, docker image and helm chart
```
Comment (alexander-falca) [inline magefile.go:36]: can we restore original comment?

File: magefile.go [hunk 3]
```diff
@@ -22,18 +22,26 @@ import (
...
 func Build() {
 	mg.SerialDeps(operator.Build)
 	mg.SerialDeps(helm.PackageChart)
 	mg.SerialDeps(dockerutils.BuildImage)
 	mg.SerialDeps(dockerutils.PushImage)
 }
+
+// RunIntegrationTests Runs Integration Tests using KUTTL
+func RunIntegrationTests() {
```
Comment (alexander-falca) [inline magefile.go:45]: we already have `integrationtests:runIntegrationTests`, do we need this target as well?

File: scale/scale_facade.go (hunk 1 above)
Comment (alexander-falca) [reply scale/scale_facade.go:215]: @yaksharma can we remove duplicate line?

File: magefile.go [hunk 4]
```diff
@@ -22,18 +22,26 @@ import (
 
 	// mage:import operator
 	"github.com/adobe/kratos/mage/operator"
+
+	// mage:import integrationTests
+	"github.com/adobe/kratos/mage/integrationTests"
 )
 
-// Cleans project
+// Clean Cleans project
```
Comment (alexander-falca) [inline magefile.go:30]: can we restore original comment?

File: mage/integrationTests/lib.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	"os"
+	"testing"
+
+	harness "github.com/kudobuilder/kuttl/pkg/apis/testharness/v1beta1"
+	"github.com/kudobuilder/kuttl/pkg/test"
+	testutils "github.com/kudobuilder/kuttl/pkg/test/utils"
+	// mage:import utils
+)
+
+// RunIntegrationTests Runs Integration Tests using KUTTL
```
Comment (alexander-falca) [inline mage/integrationTests/lib.go:26]: mage will pickup target name from func definition, we don't need to repeat it.
```suggestion
//Runs Integration Tests using KUTTL
```

File: mage/integrationTests/lib.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+func RunIntegrationTests() error {
+
+	options := harness.TestSuite{}
+
+	configPath := "tests/integration/integrationTests.yaml"
+	testToRun := "test"
+
+	// If a config is not set and kuttl-test.yaml exists, set configPath to kuttl-test.yaml.
+	if configPath == "" {
```
Comment (alexander-falca) [inline mage/integrationTests/lib.go:35]: looks like configPath is hardcoded above. Do we need this check?

File: mage/integrationTests/lib.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+		if _, err := os.Stat("kuttl-test.yaml"); err == nil {
+			configPath = "kuttl-test.yaml"
+		} else {
+			return errors.New("kuttl-test.yaml not found, provide arguments indicating the tests to load")
+		}
+	}
+
+	// Load the configuration YAML into options.
+	if configPath != "" {
```
Comment (alexander-falca) [inline mage/integrationTests/lib.go:44]: same here, configPath already defined on line 31. Do we need this if condition?

File: magefile.go (hunk 2 above)
Comment (yaksharma) [reply magefile.go]: Done.

File: scale/scale_facade.go (hunk 1 above)
Comment (yaksharma) [reply scale/scale_facade.go]: Done.

File: magefile.go (hunk 3 above)
Comment (yaksharma) [reply magefile.go]: removed the namespace and target

File: mage/integrationTests/lib.go (hunk 5 above)
Comment (yaksharma) [reply mage/integrationTests/lib.go]: Removed

**Review by alexander-falca — APPROVED** [review]

**Pull request merged**